include/boost/geometry/strategies/cartesian/side_of_intersection.hpp

```diff
@@ -27,6 +35,104 @@ namespace boost { namespace geometry
 namespace strategy { namespace side
 {
 
+namespace detail
+{
+
+template <typename T, T N, std::size_t P>
+struct integral_pow
```

Comment:
This is definitely not the best implementation of integral powers.
I will try to come up with a better one.
The problem is on the number of instantiations we have to make; it is linear in the power, whereas I believe it should be possible to do it with a logarithmic number.

I am not sure it is that important though for the purposes of this fix.